dynamodb/src/test/java/docs/javadsl/RetryTest.java

```diff
@@ -0,0 +1,203 @@
+import scala.util.Try;
+
+import java.net.URI;
+import java.time.Duration;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.*;
+
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient;
+import software.amazon.awssdk.services.dynamodb.model.BatchGetItemRequest;
+import software.amazon.awssdk.services.dynamodb.model.BatchGetItemResponse;
+import software.amazon.awssdk.services.dynamodb.model.GetItemRequest;
+import software.amazon.awssdk.services.dynamodb.model.GetItemResponse;
+
+public class RetryTest extends ItemSpecOps {
+
+  static ActorSystem system;
+  static Materializer materializer;
+  static DynamoDbAsyncClient client;
+
+  @BeforeClass
+  public static void setup() throws Exception {
+    System.setProperty("aws.accessKeyId", "someKeyId");
+    System.setProperty("aws.secretKey", "someSecretKey");
+
+    final ActorSystem system = ActorSystem.create();
+    final Materializer materializer = ActorMaterializer.create(system);
+    final DynamoDbAsyncClient client =
+        DynamoDbAsyncClient.builder()
+            .credentialsProvider(
+                StaticCredentialsProvider.create(AwsBasicCredentials.create("x", "x")))
+            .region(Region.AWS_GLOBAL)
+            .endpointOverride(new URI("http://localhost:8001/"))
+            .build();
+
+    RetryTest.system = system;
+    RetryTest.materializer = materializer;
+    RetryTest.client = client;
+  }
+
+  @AfterClass
+  public static void tearDown() {
+    StreamTestKit.assertAllStagesStopped(materializer);
+
+    client.close();
+    system.terminate();
+  }
+
+  @Before
+  public void createTable() throws Exception {
+    DynamoDb.single(client, DynamoDbOp.createTable(), createTableRequest(), materializer)
+        .toCompletableFuture()
+        .get(5, TimeUnit.SECONDS);
+  }
+
+  @After
+  public void deleteTable() throws Exception {
+    DynamoDb.single(client, DynamoDbOp.deleteTable(), deleteTableRequest(), materializer)
+        .toCompletableFuture()
+        .get(5, TimeUnit.SECONDS);
+  }
+
+  @Override
+  public String tableName() {
+    return "RetryTest";
+  }
+
+  @Test
+  public void retrySuccessfulRequests() throws Exception {
+    DynamoDb.single(
+            client, DynamoDbOp.batchWriteItem(), batchWriteLargeItemRequest(1, 25), materializer)
+        .toCompletableFuture()
+        .get(5, TimeUnit.SECONDS);
+    DynamoDb.single(
+            client, DynamoDbOp.batchWriteItem(), batchWriteLargeItemRequest(26, 50), materializer)
+        .toCompletableFuture()
+        .get(5, TimeUnit.SECONDS);
+
+    final JavaPartialFunction<
+            Pair<Try<BatchGetItemResponse>, NotUsed>,
+            akka.japi.Option<Collection<Pair<BatchGetItemRequest, NotUsed>>>>
```

Comment:
The Java code improves a tiny bit if you `import akka.japi.Option` and `import static akka.japi.Option.*` and `NotUsed.notUsed` for less need to qualify things.